torchmultimodal/modules/layers/codebook.py

```diff
@@ -195,7 +191,7 @@ def _quantize(self, encoded_flat: Tensor) -> Tuple[Tensor, Tensor]:
 
     def forward(self, z: Tensor) -> CodebookOutput:
         # First check if embedding is initialized correctly
-        if not self._is_embedding_init and self.training:
+        if torch.sum(self.embedding).item() == 0 and self.training:
```

Comment:
If you just remove `else`, we will go straight into `_preprocess` when loading the checkpoint, right?
```
if not self._is_embedding_init and self.training:
    encoded_flat, permuted_shape = self._init_embedding_and_preprocess(z)

# Reshape and flatten encoder output for quantization
encoded_flat, permuted_shape = self._preprocess(z)
```

Reply:
nope, loading the checkpoint will not switch the `_is_embedding_init` flag and it will still call `._init_embedding_and_preprocess`